src/main/java/com/aerospike/restclient/domain/RestClientIndex.java

```diff
@@ -16,125 +16,129 @@
-
-	static IndexCollectionType normalizeCollectionType(String collType) {
-		if (collType.equals("NONE")) {
-			return IndexCollectionType.DEFAULT;
-		}
-		return IndexCollectionType.valueOf(collType);
-	}
+    @JsonProperty("type")
+    IndexType indexType;
+
+    @JsonProperty("collection_type")
+    IndexCollectionType collectionType;
+
+    @Schema(example = "testNS")
+    String namespace;
+
+    @Schema(example = "testSet")
+    String set;
+
+    @Schema(description = "The bin which is indexed", example = "ageBin")
+    String bin;
+
+    @Schema(description = "The name of the index. This must be unique per set", example = "ageIndex")
+    String name;
+
+    public IndexType getIndexType() {
+        return this.indexType;
+    }
+
+    public IndexCollectionType getCollectionType() {
+        return this.collectionType == null ? IndexCollectionType.DEFAULT : this.collectionType;
+    }
+
+    public String getNamespace() {
+        return this.namespace;
+    }
+
+    public String getSet() {
+        return this.set;
+    }
+
+    public String getBin() {
+        return this.bin;
+    }
+
+    public String getName() {
+        return this.name;
+    }
+
+    public void setIndexType(IndexType iType) {
+        this.indexType = iType;
+    }
+
+    public void setCollectionType(IndexCollectionType cType) {
+        this.collectionType = cType;
+    }
+
+    public void setNamespace(String namespace) {
+        this.namespace = namespace;
+    }
+
+    public void setSet(String set) {
+        this.set = set;
+    }
+
+    public void setBin(String bin) {
+        this.bin = bin;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /* Convert an info call to a RestClientIndex */
+    @Schema(hidden = true)
+    private static final String NAME_KEY = "indexname";
+    @Schema(hidden = true)
+    private static final String NS_KEY = "ns";
+    @Schema(hidden = true)
+    private static final String SET_KEY = "set";
+    @Schema(hidden = true)
+    private static final String BIN_KEY = "bin";
+    @Schema(hidden = true)
+    private static final String INDEX_TYPE_KEY = "type";
+    @Schema(hidden = true)
+    private static final String COLLECTION_TYPE_KEY = "indextype";
+
+    public RestClientIndex() {
+    }
+
+    public RestClientIndex(Map<String, String> idxInfo) {
+        name = idxInfo.get(NAME_KEY);
+        namespace = idxInfo.get(NS_KEY);
+        set = normalizeSet(idxInfo.get(SET_KEY));
+        bin = idxInfo.get(BIN_KEY);
+        indexType = normalizeIndexType(idxInfo.get(INDEX_TYPE_KEY));
+        collectionType = normalizeCollectionType(idxInfo.get(COLLECTION_TYPE_KEY));
+    }
+
+    private String normalizeSet(String set) {
+        if (set == null || set.equals("NULL")) {
+            return null;
+        }
+        return set;
+    }
+
+    private IndexType normalizeIndexType(String idxType) {
+        idxType = idxType.toUpperCase();
+
+        if (idxType.equals("GEOJSON")) {
```

Comment:
I don't think that would work for `valueOf` which is the real problem.  In 6.1 they started returning lowercase indextypes instead of uppercase which worked perfectly with `valueOf`